File: Leetcode_solutions/search-in-a-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var searchBST = function(root, val) {
  if (!root || root.val == val) {
    return root;
  } else {
    if (val > root.val) {
      return searchBST(root.right, val);
    } else {
      return searchBST(root.left, val);
    }
  }
};

File: Leetcode_solutions/search-in-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution(object):
    def searchBST(self, root, val):
        """
        :type root: TreeNode
        :type val: int
        :rtype: TreeNode
        """
        if root and root.val == val:
            return root
        if root and val > root.val:
            return self.searchBST(root.right, val)
        if root and val < root.val:
            return self.searchBST(root.left, val)
        return None
